Userland/Utilities/pro.cpp

```diff
@@ -179,6 +184,28 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             request_headers.set(header.substring_view(0, split.value()), header.substring_view(split.value() + 1));
             return true;
         } });
+    args_parser.add_option(Core::ArgsParser::Option {
+        .argument_mode = Core::ArgsParser::OptionArgumentMode::Required,
+        .help_string = "(HTTP only) Provide basic authentication credentials (format: username:password)",
+        .long_name = "auth",
+        .short_name = 'u',
+        .value_name = "credentials",
+        .accept_value = [&](auto* s) {
+            StringView credentials { s, strlen(s) };
+            auto split = credentials.find(':');
+            // FIXME: Should empty passwords be allowed?
```

Comment:
As far as I can see, the HTTP Basic auth spec (https://www.rfc-editor.org/rfc/rfc7617.html) does not prohibit empty usernames and/or passwords.

Reply:
Ok, so judging by that, this is wrong now since we can't have either empty. Depending on what you say to the first comment above, the implementation will look a little different.

Reply:
Changed and annotated in the commit.